test/integration/utils.spec.ts

```diff
@@ -0,0 +1,14 @@
+import { isInvoiceAlreadyPaidError } from "src/utils"
+import { lndOutside1, lndOutside2, createInvoice, pay } from "test/helpers"
+
+describe("isInvoiceAlreadyPaidError", () => {
+  it("decodes error correctly", async () => {
+    const { request } = await createInvoice({ lnd: lndOutside2, tokens: 1000 })
+    await pay({ lnd: lndOutside1, request })
+    try {
+      await pay({ lnd: lndOutside1, request })
+    } catch (err) {
+      expect(isInvoiceAlreadyPaidError(err)).toBeTruthy()
```

Comment:
This is a bad way to expect an error.
- if this is just about decoding the error we should simply have a test that tests a util that does the decoding and not go through the `pay` function.
- if the `pay` function doesn't throw then the `catch` block is never executed and the test passes silently

If we really want to test that `pay` throws an exception we should use something like the [jest toThrow method](https://jestjs.io/docs/expect#tothrowerror). [Usage example in stack overflow](https://stackoverflow.com/questions/46042613/how-to-test-the-type-of-a-thrown-exception-in-jest).
But better would be to really only unit-test the decoding without the `pay` function in the call stack.

Reply:
:) really I tried to [don't change](https://github.com/GaloyMoney/galoy/blame/633e205d3194dadca4360055d1a2256d7619adae/test/integration/09g-utils.spec.ts#L13) the internals of the tests but I will fix it

Reply:
Fixed https://github.com/dolcalmi/galoy/commit/1e36a02847c49efbad52ba0c8ba738e94ddef20b and https://github.com/GaloyMoney/galoy/pull/340/commits/e8db0d20874b238ac2ce4173846763a605385ec3